contracts/programs/ocr2/src/lib.rs

```diff
@@ -563,11 +568,18 @@ fn pay_oracles_impl<'info>(
             oracle.payment_gjuels = 0;
             oracle.from_round_id = latest_round_id;
 
+            let to = if payee.owner == &token::ID {
+                payee.to_account_info()
+            } else {
+                // then the token account must be closed
+                token_receiver.to_account_info()
```

Comment:
Shouldn't this happen only on close?

In other situations it could be more efficient to skip the oracle and keep the funds in the contract (on `pay_oracles`, `accept_proposal` & `set_billing`).

Reply:
You can't skip the oracle because then the oracles aren't fully paid out so `accept_proposal`/`set_billing` won't succeed. Are you proposing to just ignore the oracle and then have the operator manually withdraw funds to resolve? That could work, but it would be easy to miss if one of the nodes didn't get paid out. 

I think I set the `token_receiver` to whoever executes the call currently so there's no extra parameter.

Reply:
I was suggesting that the owner probably doesn't expect to collect some dust every time they `set/accept_proposal` or `set_billing`, so it might go unnoticed. 

I think I'd prefer these operations to skip the invalid payee, keep the funds in the contract and then all the remaining funds are transferred out either on close or potentially an owner can withdraw on behalf of an invalid oracle (new fn).

Feels cleaner to do it that way, but maybe it's just my preference. Wdyt?